extension/scripts/features/user-alias/ttUserAlias.js

```diff
@@ -0,0 +1,85 @@
+"use strict";
+
+(async () => {
+	const feature = featureManager.registerFeature(
+		"User Alias - Chat",
+		"chat",
+		() => Object.keys(settings.userAlias).length,
+		addListeners,
+		liveReloadFunction,
+		removeAlias,
+		{
+			storage: ["settings.userAlias"],
+		},
+		null,
+		{ liveReload: true }
+	);
+
+	async function addListeners() {
+		await requireElement("#chatRoot [class*='chat-box-title_']");
+		addAliasTitle();
+		addAliasMessage();
+
+		CUSTOM_LISTENERS[EVENT_CHANNELS.CHAT_NEW].push(() => {
+			if (feature.enabled()) addAliasTitle();
+		});
+		CUSTOM_LISTENERS[EVENT_CHANNELS.CHAT_OPENED].push(() => {
+			if (feature.enabled()) {
+				addAliasTitle();
+				addAliasMessage();
+			}
+		});
+		CUSTOM_LISTENERS[EVENT_CHANNELS.CHAT_MESSAGE].push(({ message }) => {
+			if (feature.enabled()) addAliasMessage(message);
+		});
+	}
+
+	function liveReloadFunction(liveReload) {
+		if (liveReload) {
+			removeAlias();
+			addAliasTitle();
+			addAliasMessage();
+		}
+	}
+
+	function addAliasTitle() {
+		document.findAll("#chatRoot [class*='chat-box-title_']").forEach((chatTitle) => {
+			const chatPlayerTitle = chatTitle.getAttribute("title").trim();
+			if (!chatPlayerTitle ||
+			(chatPlayerTitle && ["Global", "Faction", "Company", "Trade", "People"].includes(chatPlayerTitle))
+			) return;
+
+			for (const alias of Object.values(settings.userAlias)) {
+				if (chatPlayerTitle === alias.name.trim()) {
+					const nameNode = chatTitle.find("[class*='name_']");
+					nameNode.setAttribute("data-user-name", nameNode.textContent);
```

Comment:
`data-xxx` attributes should be used with `element.dataset.xxx` (`element.dataset.userName` in this case, it will get converted to `data-user-name`).